src/elements/image/ImageElement.tsx

```diff
@@ -99,26 +103,28 @@ export const createImageFields = ({
       validators: [htmlMaxLength(250), htmlRequired()],
       placeholder: "Enter the source…",
     }),
-    role: createCustomDropdownField(undefinedDropdownValue, [
-      { text: "inline (default)", value: undefinedDropdownValue },
-      { text: "supporting", value: "supporting" },
-      { text: "showcase", value: "showcase" },
-      thumbnailOption,
-      { text: "immersive", value: "immersive" },
-    ]),
+    role: createCustomDropdownField(undefinedDropdownValue, roleOptions),
   };
 };
 
-export const createImageElement = (props: MainImageProps) =>
-  createReactElementSpec(
-    createImageFields(props),
+export const createImageElement = (options: ImageElementOptions) => {
+  const { update: updateRoleOptions, Store: RoleStore } = createStore(
+    options.additionalRoleOptions
```

Comment:
We might want to be consistent with out external facing API by calling `updateRoleOptions` -> `updateAdditionalRoleOptions`.